salt/auth/sqlite.py

```diff
@@ -0,0 +1,110 @@
+# -*- coding: utf-8 -*-
+
+'''
+Provide authentication using SQLite.
+
+When using SQLite as an authentication backend, you will need to create or
+use an existing table that has a username and a password column.
+
+To get started, create a simple table that holds just a username and
+a password. The password field will hold a SHA256 checksum (hex digest).
+
+.. code-block:: sql
+
+    CREATE TABLE `users` (
+      `id` INTEGER PRIMARY KEY ASC,
+      `username` VARCHAR(25),
+      `password` VARCHAR(64)
+    );
+
+To create a user within SQLite, execute the following statement.
+
+.. code-block:: sql
+
+    INSERT INTO users(username, password) VALUES ('diana',
+      '2bb80d537b1da3e38bd30361aa855686bde0eacd7162fef6a25fe97bf527a25b');
+
+Where username is `diana` and password is the SHA256 hex digest of `secret`.
+
+.. code-block:: yaml
+
+    sqlite_auth:
+      database: salt_users.db
```

Comment:
It's not clear where the database file is supposed to go. Since when we connect we're just passing through this value, it' not even clear in the code itself, let alone this example. This example should be modified to include an absolute path for the database file.

We should probably also include a warning that the database file should have its ownership/permissions set in such a way to prevent unauthorized access to the file.